lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1452,6 +1452,94 @@ class DecomposeAtenFlattenUsingIntsOp
 };
 } // namespace
 
+// Decompose aten.unflatten.int into aten.view op.
+namespace {
+class DecomposeAtenUnflattenIntOp
+    : public OpRewritePattern<AtenUnflattenIntOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenUnflattenIntOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value self = op.getSelf();
+    MLIRContext *context = op.getContext();
+    std::optional<unsigned> maybeRank = getTensorRank(self);
+    if (!maybeRank)
+      return rewriter.notifyMatchFailure(op, "unimplemented: unranked tensor");
+    unsigned inputRank = *maybeRank;
+    BaseTensorType inputTensorType = self.getType().cast<BaseTensorType>();
+    if (!inputTensorType.hasSizes())
```

Comment:
Check `outputTensorType` is more restrict than `inputTensorType`. Also, we should check inputTensor should have `ValueSemantics`.

Reply:
=> 
`auto inputTensorType = self.getType().cast<Torch::ValueTensorType>();`
`if (!inputTensorType || !inputTensorType.hasSizes()) {`
`  return rewriter.notifyMatchFailure(op,`
`                                         "Expected input type having sizes");`
`}`